### PR TITLE
chore: refactor tests layout

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

Refactors tesdata durectory to separate `default` tests from `cancun` and potentially more directories.

## Solution

Introduces `ForgeTestProfile` which manages configuration and layout of the test project.

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
awesome,

I believe this is exactly what we need, and unlocks the blobfee/cancun related cheatcodes

mind taking a look at this @onbjerg ?

**Review by onbjerg — APPROVED** [review]
i think this should work

**Review by mattsse — APPROVED** [review]
lgtm

File: crates/forge/tests/it/config.rs [hunk 1]
```diff
@@ -119,58 +115,59 @@ pub fn manifest_root() -> &'static Path {
...
-        .build(&PROJECT.paths.root, (*COMPILED).clone(), EVM_OPTS.local_evm_env(), opts)
+    base_runner(test_data)
+        .build(test_data.project.root(), test_data.output.clone(), opts.local_evm_env(), opts)
         .unwrap()
 }
 
 // Builds a runner that runs against forked state
-pub async fn forked_runner(rpc: &str) -> MultiContractRunner {
-    let mut opts = EVM_OPTS.clone();
+pub async fn forked_runner(test_data: &ForgeTestData, rpc: &str) -> MultiContractRunner {
```
Comment (DaniPopes) [inline crates/forge/tests/it/config.rs:159]: Can we make these methods on `ForgeTestData`?

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**